submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,211 @@
+
+![Portrait of Mike](./img/mike_stoklasa.png "A peculiar man with a comically large flask, colorized.")
+
+Many improvements could be made to this:
+
+* Dithering is only implemented on 1x1 tiles; I was having trouble adjusting the algorithm to multi-tile entities.
+* Multi-tile entities also fail to adjust the error metric property; palettization doesn't work as well when some pixels are of different size.
+* Some multi-tile entities have different rotations; a better implementation would check which orientation produces the least error before placing, instead of only placing them with the default orientation.
+* The colors themselves are hardcoded; it would be nice to dynamically load this from the game, especially with modded map colors...
+
+Despite these shortcomings, the output actually ends up pretty good for it's quick and dirty implementation. This shows that Draftsman is versatile enough for some rather eclectic purposes.
+
+![Portrait of Jay](./img/jay_bauman.png "Jay Bauman, notorious hack fraud, criminal at large. Armed and dangerous.")
+
+### Factorio Movie Player Resurrection
+
+A little while ago, while trawling for projects to use as examples for this very article, I came across the classic [Factorio Sandstorm](https://www.youtube.com/watch?v=mgfwwqwxdxY). A perfect project to adapt; or, it would have been, if I could actually load the map in-game! The original map version was still dated to the tender version `0.14.20`. In addition, a lot of the migrations that took place over such a long span of time broke the functionality of the script to convert the image frames into map data, as well as some functionality of the map itself, so just downloading an old version of Factorio and coercing it along wasn't going to be enough to revive the old save. For example, the `build.lua` script in charge of encoding still called the `automation-science-pack` and `logistic-science-pack` items `science-pack-1` and `science-pack-2`. That should give you some indication of how old the map was!
+
+It troubled me, seeing such an iconic piece of Factorio history fall into disrepair. So took the time to fix the issues and migrate it all the way to 1.1.57:
+
+| ![video of feel the beat](./img/cut_and_shrunk.gif) |
+| ------------------------ |
+| Pictured is part of [Feel the Beat](https://youtu.be/aLZQ-0dHbiU?t=38), keeping the tradition of Darude music videos. |
+
+Changing the build script could be done by hand, but the signal `raw-wood` that the machine internally used no longer exists in modern Factorio. In order to fix it, I replaced all occurrences of the signal with it in the map with `artillery-wagon` (since I knew it would be unique), and that map updating was done with a script running Draftsman code. I also added a number of other scripts for extracting the images from the source, as well as taking screenshots of the result and stitching them into an output video, which pretty easy in Python. I used said scripts to make the output above.
```

Comment:
```suggestion
Changing the build script could be done by hand, but the signal `raw-wood` that the machine internally used no longer exists in modern Factorio. In order to fix it, I replaced all occurrences of the signal with it in the map with `artillery-wagon` (since I knew it would be unique), and that map updating was done with a script running Draftsman code. I also added a number of other scripts for extracting the images from the source, as well as taking screenshots of the result and stitching them into an output video, which is pretty easy in Python. I used said scripts to make the output above.
```